src/Common/Dust.cpp

```diff
@@ -411,8 +439,14 @@ void DustSphNgbFinder<ndim, ParticleType>::FindNeibAndDoInterp
   double twork = timing->RunningTime();  // Start time (for load balancing)
 #endif
 
-  debug2("[DustSphNgbFinder::FindNeibAndDoInterp]");
-  CodeTiming::BlockTimer timer = timing->StartNewTimer("DUST_GAS_INTERPOLATE_FORCES");
+#ifdef _OPENMP
+  int Nthreads  = omp_get_max_threads() ;
+#else
+  int Nthreads  = 1 ;
+#endif
+  for (int t = neibbuf.size(); t < Nthreads; ++t)
```

Comment:
Yes, this is to avoid the memory allocations. It seemed easy enough to do it here rather than having a seperate allocation / deallocation function.